memory/jemalloc_nodump_allocator.cc

```diff
@@ -132,7 +132,9 @@ size_t JemallocNodumpAllocator::UsableSize(void* p,
 Status NewJemallocNodumpAllocator(
     JemallocAllocatorOptions& options,
     std::shared_ptr<MemoryAllocator>* memory_allocator) {
-  *memory_allocator = nullptr;
+  if (memory_allocator != nullptr) {
```

Comment:
How about checking `memory_allocator` with `nullptr` just once by combining this with another check below.
```
if (memory_allocator) {
  *memory_allocator = nullptr;
} else {
  return Status::InvalidArgument("memory_allocator must be non-null.");
}
```

Reply:
@riversand963 : Ok, I adjusted the PR once again.